server/settings/settings.go

```diff
@@ -120,8 +120,8 @@ func (s *Server) Get(ctx context.Context, q *settingspkg.SettingsQuery) (*settin
 		set.UiBannerURL = argoCDSettings.UiBannerURL
 		set.UiBannerPermanent = argoCDSettings.UiBannerPermanent
 		set.UiBannerPosition = argoCDSettings.UiBannerPosition
-		set.ControllerNamespace = s.mgr.GetNamespace()
 	}
+	set.ControllerNamespace = s.mgr.GetNamespace()
```

Comment:
Maybe instead of exposing this to everyone, we only enable it if anonymous mode is enabled?